eks_infrastructure/patch_hosts.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+#/ Usage: ./patch_hosts.sh
+
+set -ex
+source ./helper.sh
+
+# Parse parameters from build_param.json config file
+EKS_CLUSTERS=($(jq -r '.eks_clusters[]' build_param.json))
+CONTEXTS=($(jq -r '.contexts[]' build_param.json))
+EKS_VERSION=$(jq -r '.eks_version' build_param.json)
+CLUSTER_AUTOSCALAR_IMAGE_VERSION=$(jq -r '.cluster_autoscalar_image_version' build_param.json)
+
+# Create a cluster if the context is predeploy and operation is other than create
+if [ "${CONTEXTS}" = "BETA" ] && [ "${OPERATION}" != "create" ]; then
+  echo "Create a cluster inorder to perform operations other than creating cluster"
+  create_cluster
+fi
+
+# Upgrade hosts to latest ami
+upgrade_nodegroup
```

Comment:
Optionally, We can perform a validation if the cluster exists before performing this operation

Reply:
[upgrade_nodegroup](https://github.com/aws/deep-learning-containers/pull/3511/files#diff-00242329c764422d7007b79f0cf43e67bf3b2357483edf916512a0160a910470R60) already checks if the cluster already exists.